replica/tablets.cc

```diff
@@ -569,7 +567,7 @@ class tablet_incremental_selector : public sstables::incremental_selector_impl {
         auto token = pos.token();
         if (!_cur_set || pos.token() >= _lowest_next_token) {
             auto idx = _tset.group_of(token);
-            if (!token.is_maximum()) {
+            if (!token.is_maximum() && _tset._sstable_set_ids.contains(idx)) {
                 _cur_set = _tset.find_sstable_set(idx);
```

Comment:
isn't this condition wrong: _tset._sstable_set_ids.contains(idx) ?

IIRC, a previous version special cased min token (first call).

Reply:
I think this first call adjustment breaks some assumptions. If some layer does not go over the all ranges but just checks some, then I do not think it should get the next existing set.

Also before this change select returned an empty vector of sstables and with this change it is preserved (I think cleanup_state check should be added)